pkg/argocd/argocd.go

```diff
@@ -532,6 +546,45 @@ func getApplicationType(app *v1alpha1.Application) ApplicationType {
 	}
 }
 
+// getApplicationSourceType returns the source type of the application
+func getApplicationSourceType(app *v1alpha1.Application) v1alpha1.ApplicationSourceType {
+
+	if st, set := app.Annotations[common.WriteBackTargetAnnotation]; set &&
+		strings.HasPrefix(st, common.KustomizationPrefix) {
+		return v1alpha1.ApplicationSourceTypeKustomize
+	}
+
+	if app.Spec.HasMultipleSources() {
+		for _, s := range app.Status.SourceTypes {
+			if s == v1alpha1.ApplicationSourceTypeKustomize || s == v1alpha1.ApplicationSourceTypeHelm {
+				return s
+			}
+		}
+
+		log.WithContext().AddField("application", app.GetName()).Tracef("Could not determine if Source Type is Helm or Kustomize from multisource configuration. Returning first source type by default.")
+		return app.Status.SourceTypes[0]
```

Comment:
```suggestion
		for _, s := range app.Spec.Sources {
			if s.Helm != nil {
				return v1alpha1.ApplicationSourceTypeHelm
			} else if s.Kustomize != nil {
				return v1alpha1.ApplicationSourceTypeKustomize
			} else if s.Plugin != nil {
				return v1alpha1.ApplicationSourceTypePlugin
			}
		}
		return v1alpha1.ApplicationSourceTypeDirectory
```

Reply:
I tested this PR and when a multi-source application is crashed for some reason, the `status.SourceType` array is `null` and the `Status.SourceTypes[0]` call will throw an index out of range exception. The image-updater then crashes with panic. This is a proposed change that I tested and seems to work fine in our environment.

Reply:
Thank you!

I'll check this later. Usually I did not want to make changes that may affect any other behaviour, so I tried to keep most of the conditionals on the same there were already used. If these changes are working correctly, I'm more than happy to change it :).

Reply:
I added this changes, so I'll mark this conversation as resolved. Please, let me know if there is something wrong with the latest commits.